src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -480,7 +486,7 @@ public CloudSdk build() {
      * @throws AppEngineException if not found
      */
     @Nonnull
-    private Path discoverSdkPath() {
+    private Path discoverSdkPath() throws AppEngineException {
```

Comment:
Just read the other comments on this PR.  FWIW, more generally I agree with the sentiment expressed in the main thread - more explicit categories for checked exceptions would really nice.